arelle/plugin/validate/ESEF/ESEF_Current/DTS.py

```diff
@@ -147,6 +164,22 @@ def lc3wordAdjust(word: str) -> str:
                                     extLineItemsWronglyAnchored.append(modelConcept)
                     if modelConcept.isMonetary and not modelConcept.balance:
                         extMonetaryConceptsWithoutBalance.append(modelConcept)
+                    if esefDisclosureSystemYear >= 2024:
+                        widerConcept = widerNarrowerRelSet.fromModelObject(modelConcept)
+                        narrowerConcept = widerNarrowerRelSet.toModelObject(modelConcept)
+
+                        widerTypes = set(r.toModelObject.baseXbrliType for r in widerConcept)
+                        narrowerTypes = set(r.fromModelObject.baseXbrliType for r in narrowerConcept)
+
+                        if (narrowerTypes and modelConcept.baseXbrliType not in narrowerTypes) or (widerTypes and modelConcept.baseXbrliType not in widerTypes):
```

Comment:
Is `baseXbrliType` correct?  The manual seems to suggest the types need to match exactly.  Should you be able to anchor `noDecimalsMonetaryItemType` to `monetaryItemType`?

If a concept has stringItemType and the widerTypes (or narrowerTypes) is {stringItemType,monetaryItemType}, it appears this code allows that, but it seems like it shouldn't.  Should this be something like `(narrowerTypes and narrowerTypes != (conceptType,))` (etc.)?